plugins/modules/nfs.py

```diff
@@ -900,14 +900,13 @@ def validate_input(self, export_parent):
                 self.module.fail_json(msg=error_msg)
 
         # Check if valid FQDN/IP is provided
-        regex = re.compile(r'[a-zA-Z0-9_/.-:@]+$')
+        regex = re.compile(r'^[a-zA-Z0-9_/.\-:@]+$')
         for host_type in host_type_list:
+            host = host.strip()
```

Comment:
You might want to move this just after the `for host in self.module.params(host_type)` loop and just before `if regex.match(host) is None`. Right now, I believe this line will lead to an error that `host` is an undeclared variable.